test/cmd/null.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+run_kubectl_apply_null_tests(){
+    # Create a deployment
+    kubectl create -f hack/testdata/null-propagation/deployment-null.yml "${kube_flags[@]:?}"
+    # resources.limits.cpu should be nil.
+    kube::test::get_object_jsonpath_assert "deployment/my-dep-null" "{.spec.template.spec.containers[0].resources.requests.cpu}" ''
+    kube::test::get_object_jsonpath_assert "deployment/my-dep-null" "{.spec.template.spec.containers[0].resources.requests.memory}" '64Mi'
+    # The default value of the terminationMessagePolicy field is `File`, so the result will not be changed.
+    kube::test::get_object_jsonpath_assert "deployment/my-dep-null" "{.spec.template.spec.containers[0].terminationMessagePolicy}" 'File'
+
+    # kubectl apply on create should do what kubectl apply on update will accomplish.
+    kubectl apply -f hack/testdata/null-propagation/deployment-null.yml "${kube_flags[@]}"
+    kube::test::get_object_jsonpath_assert "deployment/my-dep-null" "{.spec.template.spec.containers[0].resources.requests.cpu}" ''
+    kube::test::get_object_jsonpath_assert "deployment/my-dep-null" "{.spec.template.spec.containers[0].resources.requests.memory}" '64Mi'
+    kube::test::get_object_jsonpath_assert "deployment/my-dep-null" "{.spec.template.spec.containers[0].terminationMessagePolicy}" 'File'
+
+    kubectl delete deployment my-dep-null "${kube_flags[@]}"
+    # hard.limits.cpu should be nil.
+    kubectl create -f hack/testdata/null-propagation/resourcesquota-null.yml "${kube_flags[@]}"
```

Comment:
this should be using apply, not create, to create the quota

Reply:
fixed.